Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs

```diff
@@ -60,9 +57,25 @@ public Startup(IHostingEnvironment env, ILoggerFactory loggerFactory)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            //var builder = new ContainerBuilder();
-            //builder.RegisterType<AuthMessageSender>().As<IEmailSender>().InstancePerDependency();
-            //builder.RegisterType<AuthMessageSender>().As<ISmsSender>().InstancePerDependency();
+            // Configure AppSettings using config by installing Microsoft.Extensions.Options.ConfigurationExtensions
+            //services.Configure<AppSettings>(Configuration);
+
+            //Configure EmailCrednetials using config by installing Microsoft.Extensions.Options.ConfigurationExtensions
+            services.Configure<EmailCrednetials>(myOptions =>
+            {
+                myOptions.From = Configuration["EmailCredential:From"];
+                myOptions.Host = Configuration["EmailCredential:Host"];
+                myOptions.Password = Configuration["EmailCredential:Password"];
+                myOptions.Port = Convert.ToInt32(Configuration["EmailCredential:Port"]);
+                myOptions.SslOnConnect = Convert.ToBoolean(Configuration["EmailCredential:SslOnConnect"]);
```

Comment:
This still not taking benefit of aspnet core. you are doing manual assignment where as it can do that automatically  for you.